azurelinuxagent/ga/exthandlers.py

```diff
@@ -1396,3 +1437,107 @@ def is_update_with_install(self):
         if update_mode is None:
             return True
         return update_mode.lower() == "updatewithinstall"
+
+
+class HandlerConfiguration(object):
+    def __init__(self, data):
+        if data is None:
+            raise ExtensionError('Malformed handler configuration file.')
+        if "handlerConfiguration" not in data:
+            raise ExtensionError('Malformed handler configuration file.')
+        if "linux" not in data['handlerConfiguration']:
+            raise ExtensionError('No linux configurations present in HandlerConfiguration')
+
+        self.data = data
+
+    def get_name(self):
+        return self.data["name"]
+
+    def get_version(self):
+        return self.data["version"]
+
+    def get_linux_configurations(self):
+        return self.data['handlerConfiguration']['linux']
+
+    def get_resource_configurations(self):
+        if "resources" in self.get_linux_configurations() and \
+                "cpu" in self.get_linux_configurations()["resources"] or \
+                "memory" in self.get_linux_configurations()["resources"]:
+            return self.get_linux_configurations()["resources"]
+        return None
+
+    def get_cpu_limits_for_extension(self):
+        resource_config = self.get_resource_configurations()
+
+        if resource_config and "cpu" in resource_config:
+            cpu_limits_object = CpuLimits(resource_config["cpu"])
+            return cpu_limits_object
+        else:
+            return None
+
+    def get_memory_limits_for_extension(self):
+        resource_config = self.get_resource_configurations()
+
+        if resource_config and "memory" in resource_config:
+            return MemoryLimit(resource_config["memory"])
+        else:
+            return None
+
+
+class CpuLimits(object):
+    def __init__(self, cpu_node):
+        self.cpu_limits = []
+        self.cores = []
+
+        for i in cpu_node:
+            if "cores" not in i or "limit_percentage" not in i:
+                raise ExtensionError("Malformed CPU limit node in HandlerConfiguration")
+            self.cpu_limits.append(CpuLimit(i["cores"], i["limit_percentage"]))
+            self.cores.append(i["cores"])
+
+        if DEFAULT_CORES_COUNT not in self.cores:
+            raise ExtensionError("Default CPU limit not set."
+                                 " Core configuration for {0} not present".format(DEFAULT_CORES_COUNT))
+
+        self.cpu_limits = sorted(self.cpu_limits)
+
+    def __str__(self):
+        return self.cpu_limits.__str__()
+
+
+class CpuLimit(object):
```

Comment:
naming; This will be difficult to differentiate from the other class, `CpuLimits`. maybe something like `CpuLimitInstance`?